src/wwwroot/controllers/TemplateCtrl.js

```diff
@@ -39,22 +43,46 @@
 
     function load(templateId) {
       $scope.loadInProgress = true;
-
-      return templates.getTemplateWithBody(templateId).then(function (result) {
+      return templates.getTemplate(templateId).then(function (template) {
+        switch (template.bodyType) {
+          case "empty":
+            $scope.isHtmlTemplate = !$rootScope.forceMsEditor;
+            break;
+          case "mseditor":
+            $scope.isHtmlTemplate = false;
+            break;
+          case "rawHtml":
+            $scope.isHtmlTemplate = true;
+            break;
+        }
+        return {
+          from_name: template.from_name,
+          from_email: template.from_email,
+          subject: template.subject,
+          id: template.id,
+          name: template.name,
+          body: $scope.isHtmlTemplate? templates.getTemplateBody(templateId) : null
+        };
+      }).then(function (result) {
         $scope.template.fromName = result.from_name;
         $scope.template.fromEmail = result.from_email;
         $scope.template.subject = result.subject;
         $scope.template.name = result.name;
         $scope.template.id = result.id;
-        $scope.template.content = result.body;
+        return result.body? result.body.then(function(body) {
+          return body.html;
+        }) : null;
+      }).then(function(html) {
```

Comment:
I do not understand this code.

```js
        return result.body? result.body.then(function(body) {
          return body.html;
        }) : null;
```

Reply:
result.body is null in the case where the template type is MSEditor, then if result.body is truly I will get a promise to get template body for html templates.